addons/mail/static/src/core/common/discuss_app_category_model.js

```diff
@@ -54,6 +54,16 @@ export class DiscussAppCategory extends Record {
     addTitle;
     /** @type {string} */
     addHotkey;
+    openStateKey = Record.attr("", {
+        compute() {
+            return this.serverStateKey ?? this.id;
+        },
+        onUpdate() {
+            if (this._store.settings) {
+                this._store.settings[this.openStateKey] ??= true;
+            }
+        },
+    });
```

Comment:
We should not save in settings if it's not in server settings.
Just make a new key to directly store the local value here instead.